libs/wire-api/src/Wire/API/User/Client/DPoPAccessToken.hs

```diff
@@ -0,0 +1,117 @@
+import Data.SOP
+import Data.Schema
+import Data.String.Conversions (cs)
+import qualified Data.Swagger as S
+import Data.Swagger.ParamSchema (ToParamSchema (..))
+import Data.Text as T
+import Imports
+import Servant (FromHttpApiData (..), ToHttpApiData (..))
+import Wire.API.Routes.MultiVerb
+
+newtype Proof = Proof {unProof :: ByteString}
+  deriving (Eq, Show, Generic)
+  deriving newtype (FromByteString, ToByteString)
+
+instance ToHttpApiData Proof where
+  toQueryParam = cs . toByteString'
+
+instance FromHttpApiData Proof where
+  parseQueryParam = maybe (Left "Invalid Proof") Right . fromByteString' . cs
+
+instance ToParamSchema Proof where
+  toParamSchema _ = toParamSchema (Proxy @Text)
+
+newtype DPoPAccessToken = DPoPAccessToken {unDPoPAccessToken :: ByteString}
+  deriving (Eq, Show, Generic)
+  deriving newtype (FromByteString, ToByteString)
+  deriving (FromJSON, ToJSON, S.ToSchema) via (Schema DPoPAccessToken)
+
+instance ToSchema DPoPAccessToken where
+  schema = named "DPoPAccessToken" $ unDPoPAccessToken .= fmap DPoPAccessToken base64Schema
+
+instance ToParamSchema DPoPAccessToken where
+  toParamSchema _ = toParamSchema (Proxy @Text)
+
+instance ToHttpApiData DPoPAccessToken where
+  toQueryParam = cs . toByteString'
+
+instance FromHttpApiData DPoPAccessToken where
+  parseQueryParam = maybe (Left "Invalid DPoPAccessToken") Right . fromByteString' . cs
+
+data AccessTokenType = DPoP
+  deriving (Eq, Show, Generic)
+  deriving (FromJSON, ToJSON, S.ToSchema) via (Schema AccessTokenType)
+
+instance ToSchema AccessTokenType where
+  schema =
+    enum @Text "AccessTokenType" $
+      mconcat
+        [ element "DPoP" DPoP
+        ]
+
+data DPoPAccessTokenResponse = DPoPAccessTokenResponse
+  { datrToken :: DPoPAccessToken,
+    datrType :: AccessTokenType,
+    datrExpiresIn :: Word32
+  }
+  deriving (Eq, Show, Generic)
+  deriving (FromJSON, ToJSON, S.ToSchema) via (Schema DPoPAccessTokenResponse)
+
+instance ToSchema DPoPAccessTokenResponse where
+  schema =
+    object "DPoPAccessTokenResponse" $
+      DPoPAccessTokenResponse
+        <$> datrToken .= field "token" schema
+        <*> datrType .= field "type" schema
+        <*> datrExpiresIn .= field "expires_in" schema
+
+data CacheControl = NoStore
+  deriving (Eq, Show, Generic)
+
+instance ToByteString CacheControl where
+  builder NoStore = "no-store"
+
+instance FromByteString CacheControl where
+  parser = do
+    t :: Text <- parser
+    case t & T.toLower of
+      "no-store" -> pure NoStore
+      _ -> fail $ "Invalid CacheControl type: " ++ show t
+
+instance ToHttpApiData CacheControl where
+  toQueryParam = cs . toByteString'
+
+instance FromHttpApiData CacheControl where
+  parseQueryParam = maybe (Left "Invalid CacheControl") Right . fromByteString' . cs
+
+instance ToParamSchema CacheControl where
+  toParamSchema _ = toParamSchema (Proxy @Text)
+
+instance AsHeaders '[CacheControl] DPoPAccessTokenResponse (DPoPAccessTokenResponse, CacheControl) where
```

Comment:
1. why is this defined here, but the corresp. `NonceHeader` instance is `API.Public`, see above?
2. does this mean we respond with a header here?  but we need to respond with a json body, is this code left-over from earlier plans?
3. `CacheControl` seems too general to be defined here.  can we either use untyped strings or in something like `libs/servant-utilities`?  (i'm also ok with abusing `wai-utilities` for this)